packages/lambdas/src/document-downloader.ts

```diff
@@ -0,0 +1,312 @@
+  async (req: {
+    document: DocumentWithOriginalIdAndContent;
+    fileInfo: S3Info;
+    orgName: string;
+    orgOid: string;
+    facilityNPI: string;
+  }) => {
+    const { document, fileInfo, orgName, orgOid, facilityNPI } = req;
+
+    const cwOrgCertificate: string = (await getSecret(cwOrgCertificateSecret)) as string;
+    if (!cwOrgCertificate) {
+      throw new Error(`Config error - CW_ORG_CERTIFICATE doesn't exist`);
+    }
+
+    const cwOrgPrivateKey: string = (await getSecret(cwOrgPrivateKeySecret)) as string;
+    if (!cwOrgPrivateKey) {
+      throw new Error(`Config error - CW_ORG_CERTIFICATE doesn't exist`);
+    }
+
+    const { writeStream, promise } = uploadStream(
+      fileInfo.fileName,
+      fileInfo.fileLocation,
+      document.content.mimeType
+    );
+
+    await downloadDocumentFromCW({
+      orgCertificate: cwOrgCertificate,
+      orgPrivateKey: cwOrgPrivateKey,
+      orgName,
+      orgOid,
+      facilityNPI,
+      location: document.content.location,
+      stream: writeStream,
+    });
+
+    const uploadResult = await promise;
+
+    console.log(`Uploaded ${document.id} to ${uploadResult.Location}`);
+
+    const downloadedDocument = await downloadDocumentFromS3({ fileName: uploadResult.Key });
+
+    const { size, contentType } = await getFileInfoFromS3(uploadResult.Key, uploadResult.Bucket);
+
+    const originalXml = {
+      bucket: uploadResult.Bucket,
+      key: uploadResult.Key,
+      location: uploadResult.Location,
+      size,
+      contentType,
+      isNew: true,
+      isOriginal: true,
+    };
+
+    if (downloadedDocument.data && downloadedDocument.contentType === "application/xml") {
+      const containsB64 = downloadedDocument.data.includes("nonXMLBody");
+
+      if (containsB64) {
+        const { newXML, b64 } = removeAndReturnB64FromXML(downloadedDocument.data);
+
+        const newFileName = fileInfo.fileName.split(".")[0].concat(".pdf");
+
+        const b64Buff = Buffer.from(b64, "base64");
+
+        const b64Upload = await s3client
+          .upload({
+            Bucket: bucketName,
+            Key: newFileName,
+            Body: b64Buff,
+            ContentType: "application/pdf",
+          })
+          .promise();
+
+        await s3client
+          .putObject({
+            Bucket: bucketName,
+            Key: fileInfo.fileName,
+            Body: newXML,
+            ContentType: "application/xml",
+          })
+          .promise();
+
+        const b64FileInfo = await getFileInfoFromS3(b64Upload.Key, b64Upload.Bucket);
+        const newXmlFileInfo = await getFileInfoFromS3(uploadResult.Key, uploadResult.Bucket);
+
+        originalXml.size = newXmlFileInfo.size;
+
+        const msg = `Multiple files created due to b64 in xml`;
+
+        capture.message(msg, {
+          extra: {
+            context: `documentDownloader.extractB64FromXML`,
+            b64FileName: b64Upload.Key,
+            xmlFileName: uploadResult.Key,
+          },
```

Comment:
Could add `orgName` and other relevant information to help speed the debugging process - even send `cxId` from the server?